libcxx/src/exception.cpp

```diff
@@ -6,7 +6,9 @@
 //
 //===----------------------------------------------------------------------===//
 
-#define _LIBCPP_ENABLE_CXX20_REMOVED_UNCAUGHT_EXCEPTION
+#ifndef _LIBCPP_ENABLE_CXX20_REMOVED_UNCAUGHT_EXCEPTION
+#  define _LIBCPP_ENABLE_CXX20_REMOVED_UNCAUGHT_EXCEPTION
+#endif
```

Comment:
When it's set for the whole repo globally, and passed on the command line to the compiler.

Reply:
How do you achieve this? We don't expect libc++ to be compiled with arbitrary flags like this.

Reply:
What does your CMake invocation look like?

Reply:
It's Bazel and we build several (many) projects with the same build flags. Some of them use the removed `uncaught_exception` and fail without the global flag. But with the global flag libcxx fails to build.

The projects are too many and we don't own all of them, thus we got to set the build flags globally.

Reply:
Right, I was somewhat expecting a less mainstream setup, since this shouldn't be possible if you use the normal ways to build libc++.

Libc++ does not expect to be built just like any other "user" project. It may not be safe to pass arbitrary flags you're passing when you compile user code. It's not that there's something fundamentally wrong with `_LIBCPP_ENABLE_CXX20_REMOVED_UNCAUGHT_EXCEPTION` specifically, but the code as written had the intended effect: it caught when you compiled it in a way that was unexpected. That's why I'm pushing back against this simple change: it turns something that should be caught into a silent pass, hiding the fact that your build setup is not as it should.

I don't know much about Bazel but it looks like you should make sure to isolate what flags you're passing when building projects, at least libc++.

Reply:
I understand what you mean but it's... not always practical in very big repos. I'll look for another way.

Thanks.

Reply:
I understand. Unfortunately my responsibility is to let you know what's good and what's not good to do when building libc++, but I understand that doing the right thing may not be trivial from your side. If you clean this up a bit though, it may pay off in the future by preventing libc++ from being built with unexpected flags (which could go wrong in a number of ways).